### PR TITLE
Add doc for debug solver wrapper and cleanup

### PR DESCRIPTION
**Pull request by adam-a-a**
## Summary/Motivation:
This PR adds a how-to to describe our new debug mode brought in by @bknueven in PR #1287. The how-to is largely taken from his comments in that PR.

## Changes proposed in this PR:
- add how-to
- revise some of the printouts in debug mode

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: docs/how_to_guides/how_to_use_debugging_solver_wrapper.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+Overview
+--------
+
+Debugging a failed initialization or solve can be cumbersome. 
+Further, it is often useful to have the model state (i.e., initial variable values) from before the solve failed.
+The debugging solver wrapper facilitates the following:
+
+(1) Stores initialization information before a failed solve
+
+(2) Upon a failed attempt to solve, the user is routed to an Interactive Python notebook where the restored model state can be accessed, `IDAES' DiagnosticToolbox <https://idaes-pse.readthedocs.io/en/stable/reference_guides/core/util/diagnostics/diagnostics_toolbox.html>` is instantiated to probe the model, and the user can freely apply any other diagnostic utility functions to troubleshoot the problematic model.
```
Comment (MarcusHolly) [inline docs/how_to_guides/how_to_use_debugging_solver_wrapper.rst]:
```suggestion
(2) Upon a failed attempt to solve, the user is routed to an Interactive Python notebook where the restored model state can be accessed, `IDAES' DiagnosticToolbox <https://idaes-pse.readthedocs.io/en/stable/reference_guides/core/util/diagnostics/diagnostics_toolbox.html>`_ is instantiated to probe the model, and the user can freely apply any other diagnostic utility functions to troubleshoot the problematic model. 
```
Link is not working correctly - I think you're just missing this one character

**Review by MarcusHolly — APPROVED** [review]
Just one small typo

File: docs/how_to_guides/how_to_use_debugging_solver_wrapper.rst (hunk 1 above)
Comment (adam-a-a) [reply docs/how_to_guides/how_to_use_debugging_solver_wrapper.rst]: Thanks!

**Review by ElmiraShamlou — APPROVED** [review]
LGTM

**Pull request merged**